src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,80 @@
-import { PeopleFilters } from './PeopleFilters';
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { getPeople } from '../api';
+import { Person } from '../types';
+import { getFilteredPeople, getPersonParents } from '../helpers';
+import { PeopleFilters } from './PeopleFilters';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasError, setHasError] = useState(false);
+
+  const [serachParams, setSearchParams] = useSearchParams();
+  const query = serachParams.get('query') || '';
+  const sex = serachParams.get('sex') || '';
+  const centuries = serachParams.getAll('centuries') || [];
+
+  const loadPeopleFromServer = async () => {
+    try {
+      const peopleFromServer = await getPeople();
+      const peopleWithParents: Person[] = peopleFromServer.map(
+        person => getPersonParents(person, peopleFromServer),
+      );
+
+      setPeople(peopleWithParents);
+    } catch {
+      setHasError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadPeopleFromServer();
+  }, []);
+
+  const filteredPeople = getFilteredPeople(people, query, sex, centuries);
+
   return (
     <>
       <h1 className="title">People Page</h1>
 
       <div className="block">
         <div className="columns is-desktop is-flex-direction-row-reverse">
           <div className="column is-7-tablet is-narrow-desktop">
-            <PeopleFilters />
+            <PeopleFilters
+              searchParams={serachParams}
+              onSetSearchParams={setSearchParams}
+              query={query}
```

Comment:
You don't need to pass parameters through props; instead, you can access them directly inside the component where they are needed.

Reply:
on my opinion better to pass them as props, that way we avoid code duplicating